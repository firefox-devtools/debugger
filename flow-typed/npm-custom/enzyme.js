/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// Using this custom definition of the enzyme module avoids generating some
// flow errors that are unrelated to uses of enzyme itself...
// It would be nice to clean this up.
declare module "enzyme" {
  declare module.exports: any;
}
