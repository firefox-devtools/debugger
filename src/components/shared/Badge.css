.badge {
  --size: 17px;
  --radius: calc(var(--size) / 2);
  height: var(--size);
  min-width: var(--size);
  line-height: var(--size);
  background: var(--theme-toolbar-background-hover);
  color: var(--theme-body-color);
  border-radius: var(--radius);
  padding: 0 4px;
  font-size: 0.9em;
}
