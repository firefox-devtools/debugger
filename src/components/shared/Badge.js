/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow
import React from "react";
import "./Badge.css";

type Props = {
  children: number
};

const Badge = ({ children }: Props) => (
  <span className="badge text-white text-center">{children}</span>
);

export default Badge;
