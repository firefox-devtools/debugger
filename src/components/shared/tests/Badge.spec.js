/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { shallow } from "enzyme";

import Badge from "../Badge";

describe("Badge", () => {
  it("render", () =>
    expect(
      shallow(
        <Badge>
          <p>3</p>
        </Badge>
      )
    ).toMatchSnapshot());
});
