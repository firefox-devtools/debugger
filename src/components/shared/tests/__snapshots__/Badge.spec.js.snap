// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge render 1`] = `
<span
  className="badge text-white text-center"
>
  <p>
    3
  </p>
</span>
`;
